code/WorkInProgress/MechanicMadness.dm

```diff
@@ -2072,6 +2072,69 @@ var/list/mechanics_telepads = new/list()
 		icon_state = "[under_floor ? "u":""]comp_mic"
 		return
 
+/obj/item/mechanics/radioscanner
+	name = "Radio Scanner Component"
+	desc = ""
+	icon_state = "comp_radioscanner"
+
+	var/frequency = R_FREQ_DEFAULT
+	var/datum/radio_frequency/radio_connection
+
+	get_desc()
+		. += "<br><span style=\"color:blue\">Current Frequency: [frequency]</span>"
+
+	New()
+		..()
+		mechanics.addInput("set frequency", "setfreq")
+		configs.Add(list("Set Frequency"))
+		src.append_default_configs(2)
+
+		if(radio_controller)
+			set_frequency(frequency)
+
+	attackby(obj/item/W as obj, mob/user as mob)
+		if (..(W, user)) return
+		else if (ispulsingtool(W))
+			switch (src.modify_configs())
+				if (0)
+					return
+				if ("Set Frequency")
+					var/inp = input(user, "New frequency ([R_FREQ_MINIMUM] - [R_FREQ_MAXIMUM]):", "Enter new frequency", frequency) as num
+					if (inp)
+						set_frequency(inp)
+						boutput(user, "Frequency set to [frequency]")
+
+	proc/setfreq(var/datum/mechanicsMessage/input)
+		var/newfreq = text2num(input.signal)
+		if (!newfreq) return
+		set_frequency(newfreq)
+		return
+
+	proc/set_frequency(new_frequency)
+		if (!radio_controller) return
+		new_frequency = max(R_FREQ_MINIMUM, min(new_frequency, R_FREQ_MAXIMUM))
```

Comment:
I think for regular radios you can't set even values manually if I recall correctly? This should probably behave similarly.

Reply:
Actually, you can! Regular headsets and intercoms can all be set to any frequency between R_FREQ_MINIMUM and R_FREQ_MAXIMUM. This range includes intercom channels, but no department specific frequencies. The default general chat, R_FREQ_DEFAULT (1459), falls within this range.

Reply:
When I tried I was unable to set an intercom / headset to an even frequency.

Reply:
> When I tried I was unable to set an intercom / headset to an even frequency.

IIRC, that's intentional.

Reply:
I'm dumb, I missed the word "even" in the original comment. Looks like using sanitize_frequency will handle that as well as keeping the new frequency within the allowed range. This issue should be resolved